modules/common/src/main/post-df/DownloadCsv.scala

```diff
@@ -64,7 +64,10 @@ object SpreadsheetOutput {
 
     // save as regular hadoop dir with one part-00000 file
     val temporaryDirName = s"${temporaryFileName}-tmpdir"
-    saveAsCsv(df.coalesce(1), temporaryDirName)
+
+    // df.repartition(100) needed so calculation would not happen on single node
+    // (df.coalesce(1) alone causes exception even on moderate input size)
+    saveAsCsv(df.repartition(100).coalesce(1), temporaryDirName)
```

Comment:
coalesce might happen without a shuffle, means forcing a single executor read everything in memory.